rpmlint/permissions.py

```diff
@@ -0,0 +1,202 @@
+# vim: sw=4 et sts=4 ts=4 :
+#############################################################################
+# Author        : Matthias Gerstner
+# Purpose       : reusable code for parsing permissions/chkstat profiles
+#############################################################################
+
+import copy
+import os
+
+
+class PermissionsEntry:
+
+    # source profile path
+    profile = None
+    # source profile line nr
+    linenr = None
+    # target path
+    path = None
+    owner = None
+    group = None
+    # mode as integer
+    mode = None
+    caps = []
+    # related paths from variable expansions
+    related_paths = []
+
+    def __init__(self, _profile, _line_nr):
+
+        self.profile = _profile
+        self.linenr = _line_nr
+
+    def __str__(self):
+
+        ret = '{}:{}: {path} {owner}:{group} {mode}'.format(
+            self.profile,
+            self.linenr,
+            path=self.path,
+            owner=self.owner,
+            group=self.group,
+            mode=oct(self.mode)
+        )
+
+        for cap in self.caps:
+            ret += '\n+capability ' + cap
+
+        for related in self.related_paths:
+            ret += '\nrelated to ' + related
+
+        return ret
+
+
+class VariablesHandler:
+
+    def __init__(self, variables_conf_path):
+
+        self.m_variables = {}
+
+        try:
+            with open(variables_conf_path) as fd:
+                self._parse(variables_conf_path, fd)
+        except FileNotFoundError:
+            # this can happen during migration in OBS when the new permissions
+            # package is not yet around
+            pass
+
+    def _parse(self, label, fd):
+
+        for nr, line in enumerate(fd.readlines(), 1):
+
+            line = line.strip()
+
+            if not line or line.startswith('#'):
+                continue
+
+            parts = line.split('=', 1)
+
+            if len(parts) != 2:
+                raise Exception('{}:{}: parse error'.format(label, nr))
+
+            varname = parts[0].strip()
+            values = parts[1].split()
+            # strip leading or trailing slashes
+            values = [v.strip(os.path.sep) for v in values]
+
+            self.m_variables[varname] = values
+
+    def getVariables(self):
+        """Returns a dictionary with variable names as keys and a list of
+        variable values as values."""
+        return self.m_variables
```

Comment:
Please remove it and use `self.variables` directly. The comment can be moved there.